src/main/java/eu/europa/ted/efx/interfaces/ScriptGenerator.java

```diff
@@ -368,10 +368,59 @@ public StringExpression composeSubstringExtraction(StringExpression text, Numeri
 
   public StringExpression composeToStringConversion(NumericExpression number);
 
+  /**
+   * Returns the target language script that converts the given text to upper case.
+   * 
+   * @since SDK 2.0.0
+   * @see #composeToLowerCaseConversion(StringExpression)
+   * 
+   * @param text The text to be converted to upper case.
+   * @return     The target language script that converts the text to upper case.
+   */
   public StringExpression composeToUpperCaseConversion(StringExpression text);
 
+  /**
+   * Returns the target language script that converts the given text to lower case.
+   * 
+   * @since SDK 2.0.0
+   * @see #composeToUpperCaseConversion(StringExpression)
+   * 
+   * @param text   The text to be converted to lower case.
+   * @return       The target language script that converts the text to lower case.
+   */
   public StringExpression composeToLowerCaseConversion(StringExpression text);
 
+  /**
+   * Gets the target language script that retrieves the preferred language ID
+   * out of the languages available in the given field.
+   * 
+   * The function is intended to be used in a predicate to select the text in the preferred
+   * language. The function's implementation will typically have to depend on a runtime call 
+   * to a runtime library function that retrieves the language identifiers that are preferred
+   * for the current visualisation.
+   * 
+   * @since SDK 2.0.0 
+   * @see #getTextInPreferredLanguage(PathExpression)
+   * 
+   * @param fieldReference  The multilingual text field.
+   * @return The target language script that retrieves the preferred language ID.
+   */
+  public StringExpression getPreferredLanguage(final PathExpression fieldReference);
```

Comment:
Ignore comment above, this PR is for EFX-2, so new methods are OK.
But I'm unclear on how this resolves the issue for EFX 1.x : it will only work if you translate EFX 1.x with something that uses v 2.x of the toolkit.

Reply:
The XPathScriptGeneratorV1 for EFX-1 does this implicitly. In EFX-2 you need to call a function. No implicit behaviour.